price-estimator/src/filter.rs

```diff
@@ -209,9 +273,78 @@ fn apply_rounding_buffer(amount: f64, price_rounding_buffer: f64) -> f64 {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use futures::future::FutureExt as _;
+    use anyhow::{anyhow, Result};
+    use core::orderbook::StableXOrderBookReading;
+    use futures::future::{BoxFuture, FutureExt as _};
+
+    fn empty_orderbook() -> impl StableXOrderBookReading {
```

Comment:
Is there a reason you are not using the Mocked versions here?

Reply:
Can't use mocks in other crates because they are only compiled for tests.